framework/assets/yii.gridView.js

```diff
@@ -85,25 +85,40 @@
             var $grid = $(this), event;
             var settings = gridData[$grid.attr('id')].settings;
             var data = {};
-            $.each($(settings.filterSelector).serializeArray(), function () {
-                data[this.name] = this.value;
-            });
+            var dataForm = [];
 
+            // get the query params from the request url
             $.each(yii.getQueryParams(settings.filterUrl), function (name, value) {
-                if (data[name] === undefined) {
-                    data[name] = value;
+                data[name.replace(/\[[0-9]+\]/i, '[]')] = value;
+            });
+
+            $.each($(settings.filterSelector).serializeArray(), function () {
+                if (data[this.name] !== undefined) {
+                    //unset all the values that appear on the url to avoid setting them twice.
+                    delete data[this.name];
+                }
+                if (this.value !== '') { // only add to the url values that are not empty strings
+                    dataForm.push([this.name, this.value]);
```

Comment:
I'd push objects instead of arrays:
~~~js
dataForm.push({name: this.name, value: this.value});
~~~

It's better to later reference name and value by keys instead of numeric indices.